src/resource/hal-resource-client.factory.js

```diff
@@ -201,6 +223,20 @@ export default function HalResourceClientFactory($q, $injector, $halConfiguratio
     function toStringItem(item) {
       return item.toString();
     }
+
+    /**
+     * Execute a HTTP GET request on self
+     *
+     * @param {Object|null} urlParams
+     * @param {Object}      options
+     * @return {Promise}
+     */
+    function $getSelf(urlParams, options) {
+      return $http(angular.extend({}, options, {
```

Comment:
Why don't you just use `$request('GET', 'self', null, options)` to do the request?

I don't think a self link shouldn't have `urlParams` by design. What do you think?